firefly/app.py

```diff
@@ -28,14 +29,21 @@ def __init__(self, function, **kwargs):
 
     def __call__(self, request):
         kwargs = self.get_inputs(request)
-        result = self.function(**kwargs)
-        return self.make_response(result)
+        try:
+            validate_args(self.function, kwargs)
+            result = self.function(**kwargs)
+            return self.make_response(result)
+        except ValidationError as err:
+            return self.make_response({"error": err.args[0]}, is_error=True)
+
 
     def get_inputs(self, request):
         return json.loads(request.body.decode('utf-8'))
 
-    def make_response(self, result):
+    def make_response(self, result, is_error=False):
```

Comment:
It is better to take optional `status` argument with default value as 200 instead of passing `is_error`.